doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,263 @@
+
+Custom tests (i.e. Pod manifests) would be accessible from a local disk location.  Custom test locations would be configured from within the scorecard configuration 
+file.  Custom test labels would be also in the scorecard configuration file whereas the scorecard could pick tests to run by test labels.  For example, the scorecard configuration might be as follows to indicate the presence of custom tests from different sources:
+```
+scorecard:
+  # Setting a global scorecard option
+  output: text
+  tests:
+    init-timeout: 60
+    resources:
+      limits:
+        cpu: "1"
+        memory: "200Mi"
+      requests:
+        cpu: 750m
+        memory: "100Mi"
+      test-path:
+        - "tests/here/ondisk"
+        - "someisv/tests"
+        - "operator-sdk/scorecard-tests"
+      cr-manifest:
+        - "deploy/crds/cache.example.com_v1alpha1_memcached_cr.yaml"
+        - "deploy/crds/cache.example.com_v1alpha1_memcached_cr2.yaml"
+  …
+```
+
+
+### Custom Test Execution
+
+Scorecard tests would be executed from the SDK scorecard CLI as is the case today.  Scorecard would execute the tests, capture the results, and provide feedback to the end-user by means of the current scorecard API (e.g. ScorecardTestResult).  Command line flags would allow the end-user a means to specify which tests are executed and upon which test environment to run the tests.  For example:
+```
+operator-sdk scorecard -o text --selector=suite=isvcustom
+```
+
+### Custom Test Environments
+
+Scorecard would allow you to run tests on a local Kube cluster as is the case today.  Provisioning the test cluster is external to the scope of the scorecard itself.  For example, some might want to run a scorecard on a provisioned environment such as kind (kube in docker).  
+
+### Custom Test Labeling
+As is the case today, custom tests would allow end users to specify custom labels to each test so that the scorecard could select which tests are run based on a labeling scheme.  Here is an example of user added labels to a custom test step:
+```
+apiVersion: apps/v1
+kind: Pod
+metadata:
+  name: isvtest1
+  labels:
+    vendor: someisv
+    test: isvtest1
+    suite: isvcustom
+```
+
+The end user could run that test with this command:
+```
+operator-sdk scorecard -o text --selector=test=isvtest1
+```
+
+#### Migrating Scorecard Internal Tests to Custom Test Images
+
+With the above custom test capability, there is benefit to Red Hat in migrating the existing scorecard internal tests to be based on the custom test format. 
+
+Today there are 2 internal scorecard tests that make use of the scorecard proxy, with these tests migrating to the custom test format, the scorecard proxy is likely not necessary which would also reduce the overall scorecard implementation complexity.
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
```

Comment:
I think we can remove the section so. WDYT?

Reply:
agree, will remove in next commit.